services/dynamic-scheduler/tests/unit/service_tracker/test__api.py

```diff
@@ -0,0 +1,321 @@
+# pylint:disable=redefined-outer-name
+# pylint:disable=unused-argument
+
+from collections.abc import Callable
+from datetime import timedelta
+from typing import Any, Final
+from uuid import uuid4
+
+import pytest
+from faker import Faker
+from fastapi import FastAPI
+from models_library.api_schemas_directorv2.dynamic_services import DynamicServiceGet
+from models_library.api_schemas_dynamic_scheduler.dynamic_services import (
+    DynamicServiceStart,
+    DynamicServiceStop,
+)
+from models_library.api_schemas_webserver.projects_nodes import NodeGet, NodeGetIdle
+from models_library.projects_nodes_io import NodeID
+from models_library.services_enums import ServiceState
+from pydantic import NonNegativeInt
+from pytest_simcore.helpers.typing_env import EnvVarsDict
+from servicelib.deferred_tasks import TaskUID
+from servicelib.utils import logged_gather
+from settings_library.redis import RedisSettings
+from simcore_service_dynamic_scheduler.services.service_tracker import (
+    get_all_tracked,
+    get_tracked,
+    remove_tracked,
+    set_if_status_changed,
+    set_request_as_running,
+    set_request_as_stopped,
+    set_service_status_task_uid,
+)
+from simcore_service_dynamic_scheduler.services.service_tracker._api import (
+    _LOW_RATE_POLL_INTERVAL,
+    NORMAL_RATE_POLL_INTERVAL,
+    _get_current_state,
+    _get_poll_interval,
+)
+from simcore_service_dynamic_scheduler.services.service_tracker._models import (
+    SchedulerServiceState,
+    UserRequestedState,
+)
+
+pytest_simcore_core_services_selection = [
+    "redis",
+]
+
+
+@pytest.fixture
+def app_environment(
+    disable_rabbitmq_setup: None,
+    disable_deferred_manager_setup: None,
+    disable_notifier_setup: None,
+    app_environment: EnvVarsDict,
+    redis_service: RedisSettings,
+    remove_redis_data: None,
+) -> EnvVarsDict:
+    return app_environment
+
+
+@pytest.fixture
+def node_id() -> NodeID:
+    return uuid4()
```

Comment:
this fixture exists already at several places. please reuse that?

Reply:
TIP: check pytest_simcore's plugins called `faker_*_data` ... add them in the `pytest_plugins` list first
![image](https://github.com/user-attachments/assets/0fe21d13-5b69-4310-9fdc-d6185b0daf26)